src/Grpc.AspNetCore.Server/Internal/ClientStreamingServerCallHandler.cs

```diff
@@ -53,10 +53,16 @@ public override async Task HandleCallAsync(HttpContext httpContext)
             var service = activator.Create();
             var serverCallContext = new HttpContextServerCallContext(httpContext);
 
-            var response = await _invoker(
-                service,
-                new HttpContextStreamReader<TRequest>(httpContext, Method.RequestMarshaller.Deserializer),
-                serverCallContext);
+            serverCallContext.Initialize();
+
+            TResponse response;
+            using (serverCallContext)
+            {
+                response = await _invoker(
+                    service,
+                    new HttpContextStreamReader<TRequest>(httpContext, Method.RequestMarshaller.Deserializer),
+                    serverCallContext);
+            }
```

Comment:
This is fine, though I could also imagine allowing cancellation up until the point where we are done serializing the response and are about the write to the body.